.github/workflows/setup-test-workspace/action.yml

```diff
@@ -0,0 +1,24 @@
+name: Setup test workspace
+inputs:
+  ref-slice-ref:
+    description: 'Reference of the ref_slice to checkout'
```

Comment:
Consider being explicit about this being a *git* reference: a branch, tag, or commit hash. `Reference` is a generic term so this description doesn't communicate much beyond what the input name already got across.